app/src/main/java/com/example/marvelapp/usecase/GetCharactersUseCaseImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.example.marvelapp.usecase
+
+import com.example.marvelapp.api.service.CharacterService
+import com.example.marvelapp.api.util.Result
+import com.example.marvelapp.entity.Character
+import org.koin.core.component.KoinComponent
+
+interface GetCharactersUseCase : KoinComponent {
+    operator fun invoke(): Result<List<Character>>
+}
+
+class GetCharactersUseCaseImpl(private val charactersService: CharacterService) : GetCharactersUseCase {
+
+    override operator fun invoke(): Result<List<Character>> {
+        return when (val serviceResult = charactersService.getCharacters()) {
```

Comment:
is this logic necessary? I think that you can return directly the service like

`override operator fun invoke(): Result<List<Character>> = charactersService.getCharacters()`

Reply:
Yes, I think is a better option. Changed